### ISSUE TITLE
Sentiment analysis failures: invalid device function

### ISSUE THREAD

**Issue by elvinpoon** [baidu/Paddle#34]
Here is the error code.
```
./train.sh
I0903 14:10:57.917793 18690 Util.cpp:144] commandline: /data2/package/pypaddle/bin/../opt/paddle/bin/paddle_trainer --config=trainer_config.py --save_dir=./model_output --job=train --use_gpu=1 --trainer_count=4 --num_passes=10 --log_period=10 --dot_period=20 --show_parameter_stats_period=100 --test_all_data_in_one_period=1
I0903 14:11:01.704715 18690 Util.cpp:113] Calling runInitFunctions
I0903 14:11:01.705032 18690 Util.cpp:126] Call runInitFunctions done.
[INFO 2016-09-03 14:11:02,367 networks.py:1122] The input order is [word, label]
[INFO 2016-09-03 14:11:02,368 networks.py:1129] The output order is [__cost_0__]
I0903 14:11:02.395427 18690 Trainer.cpp:169] trainer mode: Normal
I0903 14:11:02.395754 18690 MultiGradientMachine.cpp:108] numLogicalDevices=1 numThreads=4 numDevices=4
F0903 14:11:02.400593 18690 hl_gpu_matrix_kernel.cuh:181] Check failed: cudaSuccess == err (0 vs. 8) [hl_gpu_apply_unary_op failed] CUDA error: invalid device function
*** Check failure stack trace: ***
    @     0x7fc9c37175cd  google::LogMessage::Fail()
    @     0x7fc9c3719433  google::LogMessage::SendToLog()
    @     0x7fc9c371715b  google::LogMessage::Flush()
    @     0x7fc9c3719e1e  google::LogMessageFatal::~LogMessageFatal()
    @           0x7d65b2  hl_gpu_apply_unary_op<>()
    @           0x79d156  paddle::BaseMatrixT<>::applyUnary<>()
    @           0x79ccf0  paddle::BaseMatrixT<>::applyUnary<>()
    @           0x780733  paddle::BaseMatrixT<>::zero()
    @           0x561960  paddle::Parameter::enableType()
    @           0x564531  paddle::parameterInitNN()
    @           0x567fe9  paddle::NeuralNetwork::init()
    @           0x55ee4b  paddle::TrainerThread::TrainerThread()
    @           0x55fab7  paddle::MultiGradientMachine::MultiGradientMachine()
    @           0x58788e  paddle::GradientMachine::create()
    @           0x6e296d  paddle::TrainerInternal::init()
    @           0x6dc144  paddle::Trainer::init()
    @           0x54622d  main
    @     0x7fc9c2699830  __libc_start_main
    @           0x54db19  _start
    @              (nil)  (unknown)
/data2/package/pypaddle/bin/paddle: line 46: 18690 Aborted                 ${DEBUGGER} $MYDIR/../opt/paddle/bin/paddle_trainer ${@:2}
```

**Comment by gangliao** [comment]
Looks like you use cuda 8.0 with modern Gpu.

invalid device function indicates that you have a CUDA / GPU incompatibility.
Maybe you can modify CMake to fix it.

open `cmake/flags.cmake` and add following code:

`if (CUDA_VERSION VERSION_GREATER "8.0")`

`list(APPEND __arch_flags " -gencode arch=compute_60,code=sm_60")`

`endif()`

then, rebuild the project

**Comment by elvinpoon** [comment]
I tried this but it doesn't work...same error

**Comment by gangliao** [comment]
Fix CUDA_VERSION Comparsion #165
